napari/utils/colormaps/_tests/test_colormaps.py

```diff
@@ -235,3 +235,33 @@ def test_ensure_colormap_with_multi_colors(colors):
     expected_colors = transform_color(colors)
     np.testing.assert_array_equal(colormap.colors, expected_colors)
     assert re.match(r'\[unnamed colormap \d+\]', colormap.name) is not None
+
+
+def test_ensure_colormap_with_hex_color_string():
+    """
+    Test all the accepted hex color representations (single/double digit rgb with/without alpha)
+    """
+
+    for color in ['#abc', '#abcd', '#abcdef', '#00ABCDEF']:
+        cmap = ensure_colormap(color)
+        assert isinstance(cmap, Colormap)
+        assert cmap.name == 'custom-' + color.lower()
+
+
+def test_ensure_colormap_with_recognized_hex_color_string():
```

Comment:
Would be nicer to have these tests use `parametrize` instead of a loop, but that's not a blocker for merging :)

Reply:
> Would be nicer to have these tests use `parametrize` instead of a loop,

No worries, that was an easy change to make :)

And I added one extra test to make sure that when a matplotlib colour name (like 'chartreuse') is passed to ensure_colormap(), the name of the generated colormap is identical to that.